### PR TITLE
[4.4.01] Add SIMD AVX2 mask assignment fix and update changelog for 4.4.01

### PR DESCRIPTION
**Pull request by crtrott**
Cherry pick of #7296

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,9 +5,10 @@
 
 ### Feature:
 * Introduce new SequentialHostInit view allocation property [\#7229](https://github.com/kokkos/kokkos/pull/7229)
+* Experimental CUDA support for unified memory mode (intended for Grace-Hopper etc.) [\#6823](https://github.com/kokkos/kokkos/pull/6823)
 
 ### Bug Fix
-* Fix issue related to the visibility of an internal symbol with shared libraries that affected `ScatterView` in particular [\#7284](https://github.com/kokkos/kokkos/pull/7284)
+* Fix issue related to the visibility of an internal OpenMP symbol with shared libraries that affected `ScatterView` in particular [\#7284](https://github.com/kokkos/kokkos/pull/7284)
```
Comment (dalg24) [inline CHANGELOG.md:11]: How about this instead?
```suggestion
* OpenMP: Fix issue related to the visibility of an internal symbol with shared libraries that affected `ScatterView` in particular [\#7284](https://github.com/kokkos/kokkos/pull/7284)
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,9 +5,10 @@
 
 ### Feature:
 * Introduce new SequentialHostInit view allocation property [\#7229](https://github.com/kokkos/kokkos/pull/7229)
+* Experimental CUDA support for unified memory mode (intended for Grace-Hopper etc.) [\#6823](https://github.com/kokkos/kokkos/pull/6823)
```
Comment (dalg24) [inline CHANGELOG.md:8]: Not sure if it deserves to be a feature considering it is an IMP_ option
Maybe "backend enhancement"?

Comment (crtrott) [reply CHANGELOG.md:8]: yeah sure

Comment (crtrott) [reply CHANGELOG.md]: I moved it

File: CHANGELOG.md [hunk 3]
```diff
@@ -6,8 +6,12 @@
 ### Feature:
 * Introduce new SequentialHostInit view allocation property [\#7229](https://github.com/kokkos/kokkos/pull/7229)
 
+### CUDA:
+* Experimental support for unified memory mode (intended for Grace-Hopper etc.) [\#6823](https://github.com/kokkos/kokkos/pull/6823)
+
 ### Bug Fix
```
Comment (dalg24) [inline CHANGELOG.md:12]: It is clashing with the way we usually organize our change log

Comment (crtrott) [reply CHANGELOG.md:12]: But its the same as we did for MI300 stuff ;-)

Comment (dalg24) [reply CHANGELOG.md:12]: It had a `Backend and Architecture Enhancements:` section header

**Review by dalg24 — APPROVED** [review]

**Review by ldh4 — APPROVED** [review]

**Pull request merged**